src/NexusMods.App.UI/Pages/ModLibrary/FileOrigins/FileOriginsPageView.axaml.cs

```diff
@@ -16,8 +23,37 @@ public FileOriginsPageView()
                     vm => vm.FileOrigins,
                     v => v.DataGrid.ItemsSource)
                 .DisposeWith(d);
-            
+
+            DataGrid.SelectedItemsToViewModelObservableChangeSetProperty(ViewModel!, vm => vm.SelectedModsObservable)
+                .DisposeWith(d);
+
+            // Synchronize the Grid with DataModel for Selected Items
+            // Enable/Disable Add Mod & Add Mod Advanced Buttons
+            void UpdateAddButtonState(bool isModAdded)
+            {
+                var enable = false;
+                if (!isModAdded)
+                    enable = ViewModel!.SelectedModsCollection.Count > 0 && 
+                             !ViewModel!.SelectedModsCollection.Any(x => x.IsModAddedToLoadout);
+
+                AddModButton.IsEnabled = enable;
+                // Note Add(Advanced) should always be available.
+            }
```

Comment:
Use CanExecute of the Commands, put commands in VM. 
Both Add and Add advanced should be enabled if there is any selection. Doesn't matter if that mod was already installed.
That is overcomplicating the code and is isn't the desired behavior, please just get rid of it.